src/main/java/Deadline.java

```diff
@@ -0,0 +1,45 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a Task Object that has a certain deadline to be met.
+ */
+public class Deadline extends Task {
+    protected String by;
+    protected LocalDate byDate;
+
+    /**
+     * Constructor for this Deadline object.
+     * @param   description  Task Description.
+     * @param   by           Due date of task.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+        this.byDate = parseDate(by);
+    }
+
+
+    private LocalDate parseDate(String str) {
+        return LocalDate.parse(str);
+    }
+
+    /**
+     * Converts information about the task to be saved to hard disk.
+     * @return string containing information about the task.
+     */
+    @Override
+    public String save() {
+        return "[D]" + super.toString() + "(by: " + by + ")";
+    }

```

Comment:
maybe can add return type of the function into the javadoc? The prob exits in multiple parts of the PRs i think, if am not wrong